runtime/src/bank.rs

```diff
@@ -135,15 +140,22 @@ impl Bank {
         bank.parent = RwLock::new(Some(parent.clone()));
         bank.parent_hash = parent.hash();
         bank.collector_id = collector_id;
+        bank.accounts = Some(parent.accounts());
+        bank.accounts().new_from_parent(bank.id, parent.id);
 
         bank
     }
 
     /// Create a new bank that points to an immutable checkpoint of another bank.
     /// TODO: remove me in favor of _and_id(), id should not be an assumed value
     pub fn new_from_parent(parent: &Arc<Bank>) -> Self {
+        static BANK_ID: AtomicUsize = AtomicUsize::new(1);
```

Comment:
is this because ID figures in the path?

Reply:
Well it figures in the account set, so it has to be unique because the accounts store is shared between all banks in the system.

Reply:
It kind of hijacks this id, which was one of my concerns. Does it ruin it for what you wanted to do with it?

Reply:
I only added the += 1 stuff for tests.  all other callers are calling new_from_parent_with_id()...